### PR TITLE
Preferences will affect day

### PR DESCRIPTION
**Pull request by AndyS120**
### Changes:

- Added function to send preferences to backend LLM
- Added an _info message_ that displays progress when sending preferences to backend.
- When adding tags, you can now press <kbd>Enter</kbd> to add tags.
- Selected nodes show their respective _types/tags_.
- Since requests to the Places API can only take one _type_, multiple requests are now being made.
  - This creates an inconvenience where these requests finish at different times, and the page doesn't wait for them to finish.
  - _This has been fixed but could be **improved**_.
- Each LLM tag received corresponds to a _place_.
  - Sending the LLM tags like `food` result in multiple tags returned.

> [!Warning]
> Some variables and functions were renamed.
> 
> | Previous | Renamed to |
> | --- | --- |
> |`makeRequest()`|`getTestMessageFromAPI()`|
> |`data`|`tripData`|
> |`request`|`directionsRequest`|
> |`request`|`nearbySearchRequest`|

### CODE REVIEW

**Comment by wrycode** [comment]
Overall, really nice work. However, I think there is a bug where it only displays one destination

**Comment by AndyS120** [comment]
I forgot to mention that each tag corresponds to a place. The LLM may also return multiple tags for a single input (try `food`). If no place was found, `console.error` will let you know.

**Review by benjaminbeach01 — COMMENTED** [review]
I cannot get more than one node to appear upon creating a trip, whether I add tags or not. This is potentially a CORS issue but I am uncertain.

**Comment by benjaminbeach01** [comment]
It is working as you have described it on the deployed webapp, I am getting results similar to your previous comment, that being that there is at least one suggested place per tag.

How feasible is it to get it to make a list of places like before that are related to the tags, as opposed to one place per tag? That way we can make a list that is feasible to a daytrip regardless of how many tags are added. I know you said that some tags will give multiple responses, though that ends up making a trip with a total length of over 15 hours.

Basically, how possible is it to take the tags, and force the response to have a minimum and maximum amount of locations suggested?

**Comment by AndyS120** [comment]
Updated PR #28
